### PR TITLE
Integration Test framework improvements

### PR DESCRIPTION
**Pull request by AntonAndell**

### CODE REVIEW

File: test-lib/src/main/java/network/balanced/score/lib/test/integration/BalancedClient.java [hunk 1]
```diff
@@ -54,6 +81,12 @@ public BalancedClient(Balanced balanced, KeyWallet wallet) {
         loans = new LoansScoreClient(chain.getEndpointURL(), chain.networkId, wallet, balanced.loans._address());
         baln = new BalnScoreClient(chain.getEndpointURL(), chain.networkId, wallet, balanced.baln._address());
         rebalancing = new RebalancingScoreClient(chain.getEndpointURL(), chain.networkId, wallet, balanced.rebalancing._address());
+        sicx = new SicxScoreClient(chain.getEndpointURL(), chain.networkId, wallet, balanced.sicx._address());
+        dex = new DexScoreClient(chain.getEndpointURL(), chain.networkId, wallet, balanced.dex._address());
+        stability = new StabilityScoreClient(chain.getEndpointURL(), chain.networkId, wallet, balanced.stability._address());
+        stakedLp = new StakedLPScoreClient(chain.getEndpointURL(), chain.networkId, wallet, balanced.stakedLp._address());
+        dividends = new DividendsScoreClient(chain.getEndpointURL(), chain.networkId, wallet, balanced.dividends._address());
+        systemScore = new SystemInterfaceScoreClient(chain.getEndpointURL(), chain.networkId, wallet, balanced.dividends._address());
```
Comment (nightowl121) [inline test-lib/src/main/java/network/balanced/score/lib/test/integration/BalancedClient.java:89]: Why is the system score set up with dividends address ??

Comment (AntonAndell) [reply test-lib/src/main/java/network/balanced/score/lib/test/integration/BalancedClient.java:89]: No reason, its just very wrong. Guess i have not used that from the client yet. Good catch

File: test-lib/src/main/java/network/balanced/score/lib/test/integration/Balanced.java [hunk 2]
```diff
@@ -52,80 +55,82 @@ public class Balanced {
...
-        daofund = deploy(owner, "DAOfund", Map.of("_governance", governance._address()));
-        dividends = deploy(owner, "Dividends", Map.of("_governance", governance._address()));
-        oracle = deploy(owner, "Oracle",null);
-        reserve = deploy(owner, "Reserve", Map.of("governance", governance._address()));
-        router = deploy(owner, "Router", Map.of("_governance", governance._address()));
-        stakedLp = deploy(owner, "StakedLP", Map.of("governance", governance._address()));
+    public void deployContracts() {
+        governance = deploy(owner, "Governance", null);
+
+        Hash balnTx = deployAsync(owner, "Baln", Map.of("_governance", governance._address()));
```
Comment (Suyog007) [inline test-lib/src/main/java/network/balanced/score/lib/test/integration/Balanced.java:87]: I am unable to deploy any of the contracts. 
It says 
```
'foundation.icon.jsonrpc.model.Hash foundation.icon.score.client.DefaultScoreClient._deployAsync(java.lang.String, java.math.BigInteger, foundation.icon.icx.Wallet, java.lang.String, java.util.Map)'
java.lang.NoSuchMethodError: 'foundation.icon.jsonrpc.model.Hash foundation.icon.score.client.DefaultScoreClient._deployAsync(java.lang.String, java.math.BigInteger, foundation.icon.icx.Wallet, java.lang.String, java.util.Map)'
```

Is there anything that i need to set before calling deployAsync?

Comment (AntonAndell) [reply test-lib/src/main/java/network/balanced/score/lib/test/integration/Balanced.java:87]: The grader in the contract needs to use our version of score-client. Check Governance for example. Think this is why

Comment (nightowl121) [reply test-lib/src/main/java/network/balanced/score/lib/test/integration/Balanced.java:87]: yeah that's the reason. it has been resolved.

**Pull request merged**